### PR TITLE
[RLlib] Discussion 2022: Fix batch_mode="complete_episodes" documentation inaccuracy.

### PR DESCRIPTION
**Pull request by sven1977**
Fix batch_mode="complete_episodes" documentation inaccuracy.

See also this discussion here:
https://discuss.ray.io/t/rllib-batch-size-for-complete-episodes-issue/2022/4

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/rllib/rllib-sample-collection.rst [hunk 1]
```diff
@@ -37,10 +37,12 @@ are determined by the following Trainer config keys:
...
-     Each rollout is exactly one episode long and always starts
-     at the beginning of an episode. It does not matter how long an episode lasts.
-     The ``rollout_fragment_length`` setting will be ignored. Note that you have to be
-     careful when chosing ``complete_episodes`` as batch_mode: If your environment does not
+     Each rollout always only contains completed episodes (at least one, but possibly more) and always starts
+     at the beginning of an episode. The ``rollout_fragment_length`` setting defines the minimum number of
+     timesteps that will be covered in the rollout. For example, if ``rollout_fragment_length=100`` and your episodes
+     are always 98 timesteps long, all rollouts will happen over two complete episodes
+     (98 < 100 -> too short; 98+98 >= 100 -> good).
+     Note that you have to be careful when chosing ``complete_episodes`` as batch_mode: If your environment does not
```
Comment (avnishn) [inline doc/source/rllib/rllib-sample-collection.rst:45]: I think that chosing should instead be -> choosing

File: doc/source/rllib/rllib-sample-collection.rst [hunk 2]
```diff
@@ -37,10 +37,12 @@ are determined by the following Trainer config keys:
...
     *complete_episodes*:
-     Each rollout is exactly one episode long and always starts
-     at the beginning of an episode. It does not matter how long an episode lasts.
-     The ``rollout_fragment_length`` setting will be ignored. Note that you have to be
-     careful when chosing ``complete_episodes`` as batch_mode: If your environment does not
+     Each rollout always only contains completed episodes (at least one, but possibly more) and always starts
+     at the beginning of an episode. The ``rollout_fragment_length`` setting defines the minimum number of
+     timesteps that will be covered in the rollout. For example, if ``rollout_fragment_length=100`` and your episodes
+     are always 98 timesteps long, all rollouts will happen over two complete episodes
+     (98 < 100 -> too short; 98+98 >= 100 -> good).
```
Comment (avnishn) [inline doc/source/rllib/rllib-sample-collection.rst:44]: does this mean that the fragment that is returned is 196 timesteps long, or that we take 2 timesteps from the second episode and append them with the first episode when returning the rollout fragment? I think we should document the answer to this question.

**Review by avnishn — COMMENTED** [review]
I have a question and a grammar fix

File: doc/source/rllib/rllib-sample-collection.rst (hunk 2 above)
Comment (sven1977) [reply doc/source/rllib/rllib-sample-collection.rst:44]: Good point! The returned fragment is indeed 196 timesteps long (2 *complete* episodes). I'll clarify this in the text.

Comment (sven1977) [inline doc/source/rllib/rllib-sample-collection.rst:44]:
```suggestion
     Each rollout always only contains **full** episodes (from beginning to terminal), never any episode fragments. The number of episodes in the rollout is 1 or larger.
     The ``rollout_fragment_length`` setting defines the minimum number of
     timesteps that will be covered in the rollout.
     For example, if ``rollout_fragment_length=100`` and your episodes are always 98 timesteps long, then rollouts will happen over two complete episodes and always be 196 timesteps long: 98 < 100 -> too short, keep rollout going; 98+98 >= 100 -> good, stop rollout after 2 episodes (196 timesteps).
```

File: doc/source/rllib/rllib-sample-collection.rst (hunk 1 above)
Comment (sven1977) [inline doc/source/rllib/rllib-sample-collection.rst:45]:
```suggestion
     Note that you have to be careful when choosing ``complete_episodes`` as batch_mode: If your environment does not
```

**Review by avnishn — APPROVED** [review]
This is crystal clear

**Pull request merged**